### DISCUSSION TITLE
1BRC in Erlang

### DISCUSSION THREAD

**Discussion by jesperes** [gunnarmorling/1brc#396] [Show and tell]
[Repository](https://github.com/jesperes/erlang_1brc/blob/main/src/aggregate.erl).

EDIT (updated 2024-01-27):

Originally ran in 130s on my i7-10610U laptop, but here are the updated numbers.

| Time  | Hardware  | Logical processors |
|-------|-----------|--------------------|
| 13.5s | i5-12400F | 12                 |
| 36.0s | i7-10610U | 8                  |

**Comment by Kartstig** [comment]
https://github.com/Kartstig/1brc_erl

Runs in 80s on my Threadripper 1950X

**Comment by onno-vos-dev** [comment]
[1brc](https://github.com/onno-vos-dev/1brc)

Runs in 0:27.37 in my Mac M1 Pro

**Comment by jesperes** [comment]
After several sets of improvements:
1. @garazdawi optimized away a binary copy and added flow-control
2. Match-context optimizations
3. Pre-computed/compressed keys

This now runs on 13.5s on my i5-12400F.
